### PR TITLE
fix(uniswapx-sdk): update known errors with priority order reactor ones

### PR DESCRIPTION
**Pull request by zhongeric**
## PR Scope

Please title your PR according to the following types and scopes following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/):

- `fix(SDK name):` will trigger a patch version
- `chore(<type>):` will not trigger any release and should be used for internal repo changes
- `<type>(public):` will trigger a patch version for non-code changes (e.g. README changes)
- `feat(SDK name):` will trigger a minor version
- `feat(breaking):` will trigger a major version for a breaking change

## Description

_[Summary of the change, motivation, and context]_

## How Has This Been Tested?

_[e.g. Manually, E2E tests, unit tests, Storybook]_

## Are there any breaking changes?

_[e.g. Type definitions, API definitions]_

If there are breaking changes, please ensure you bump the major version Bump the major version (by using the title `feat(breaking): ...`), post a notice in #eng-sdks, and explicitly notify all Uniswap Labs consumers of the SDK.

## (Optional) Feedback Focus

_[Specific parts of this PR you'd like feedback on, or that reviewers should pay closer attention to]_

## (Optional) Follow Ups

_[Things that weren't addressed in this PR, ways you plan to build on this work, or other ways this work could be extended]_

### CODE REVIEW

File: sdks/uniswapx-sdk/src/utils/OrderQuoter.ts [hunk 1]
```diff
@@ -161,7 +171,7 @@ export class UniswapXOrderQuoter
   protected quoter: OrderQuoterContract;
 
   constructor(
-    protected provider: BaseProvider,
+    protected provider: JsonRpcProvider,
```
Comment (ConjunctiveNormalForm) [inline sdks/uniswapx-sdk/src/utils/OrderQuoter.ts:174]: why the change to JsonRpcProvider only?

**Review by ConjunctiveNormalForm — APPROVED** [review]
one comment

File: sdks/uniswapx-sdk/src/utils/OrderQuoter.ts (hunk 1 above)
Comment (zhongeric) [reply sdks/uniswapx-sdk/src/utils/OrderQuoter.ts:174]: so with JsonRpcProvider we can send raw rpc calls which allows us to also optionally use the stateOverrides and blockOverrides fields

**Review by marktoda — APPROVED** [review]
I think theres a set of tests that throw the errors and make sure they capture properly

**Review by ConjunctiveNormalForm — APPROVED** [review]

**Review by ConjunctiveNormalForm — APPROVED** [review]

**Comment by zhongeric** [comment]
> I think theres a set of tests that throw the errors and make sure they capture properly

Yea, am going to add integ tests in this PR #73
